### DISCUSSION TITLE
How better to handle fork of zombienet feature?

### DISCUSSION THREAD

**Discussion by dzmitry-lahoda** [paritytech/zombienet#957] [Q&A] [answered]
I want feature of fork-of-substrate in zombienet.  To run from state of real chain.

How should I approach this imlementation? Can it be purely external to zombienet, so I just download state snapshot, make it more "local", and just feed into zombient? What would be config parameters? 

Will it require some modifications to zombienet?

**Comment by pepoviola** [comment, marked as answer]
Hi @dzmitry-lahoda, this is a feature that we will want to add at some point. I think the best approach is to use  a combinations of steps to get the fork working. At first get the state and feed into zombienet to customize the chain-spec with the needed info to make it works (e.g keys / paras / etc ), then spawn the desired network and wire everything together.

I will ping you when we have a PoC.

Thanks!

**Comment by dzmitry-lahoda** [comment]
there is 2 tools to maintain forking, so no need more.
